src/tests/EShopworld.WorkerProcess.UnitTests/Stores/CosmosDbLeaseStoreTests.cs

```diff
@@ -204,6 +204,110 @@ public async Task TestTryCreateLeaseAsyncConflict()
             result.Result.Should().BeFalse();
         }
 
+        [Fact, IsUnit]
+        public async Task TestAddLeaseRequestAsync_WhenAddingLeaseRequest_ShouldReturnTrue()
+        {
+            // Arrange
+            _mockDocumentClient.Setup(m => m.CreateDocumentAsync(
+                    It.Is<Uri>(u => u == UriFactory.CreateDocumentCollectionUri(_options.Database, "LeaseRequests")),
+                    It.IsAny<object>(),
+                    It.IsAny<RequestOptions>(),
+                    It.IsAny<bool>(),
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new Document().ToResourceResponse(HttpStatusCode.Created));
+
+            // Act
+            var result = await _store.AddLeaseRequestAsync("worktype", 1, Guid.NewGuid());
+
+            // Assert
+            result.Should().BeTrue();
+            _mockTelemetry.Verify(tel=>tel.Publish(It.IsAny<ExceptionEvent>(),It.IsAny<string>(),
+                It.IsAny<string>(),It.IsAny<int>()),Times.Never);
+
+        }
+
+        [Fact, IsUnit]
+        public async Task TestAddLeaseRequestAsync_WhenAddingLeaseRequestWithConflict_ShouldReturnFalse()
+        {
+            // Arrange
+            _mockDocumentClient.Setup(m => m.CreateDocumentAsync(
+                    It.Is<Uri>(u => u == UriFactory.CreateDocumentCollectionUri(_options.Database, _options.RequestsCollection)),
+                    It.IsAny<object>(),
+                    It.IsAny<RequestOptions>(),
+                    It.IsAny<bool>(),
+                    It.IsAny<CancellationToken>()))
+                .ThrowsAsync(CreateDocumentClientExceptionForTesting(new Error(), HttpStatusCode.Conflict));
+
+            // Act
+            var result = await _store.AddLeaseRequestAsync("worktype", 1, Guid.NewGuid());
+
+            // Assert
+            result.Should().BeFalse();
+            _mockTelemetry.Verify(tel => tel.Publish(It.IsAny<ExceptionEvent>(), It.IsAny<string>(),
+                It.IsAny<string>(), It.IsAny<int>()), Times.Once);
+
+        }
+
+        [Fact, IsUnit]
+        public async Task SelectWinnerRequestAsync_WhenMultipleRequests_ShouldReturnTheOneWithHighestPriority()
+        {
+            // Arrange
+            var winnerLeaseRequest = new LeaseRequest
+            {
+                Id = Guid.NewGuid().ToString(),
+                InstanceId = Guid.NewGuid(),
+                LeaseType = "leasetype",
+                Priority = 0
+            };
+            var dataSource = new List<LeaseRequest>
+            {
+                new LeaseRequest
+                {
+                    Id = Guid.NewGuid().ToString(),
+                    InstanceId = Guid.NewGuid(),
+                    LeaseType = "leasetype",
+                    Priority = 1
+                },
+                winnerLeaseRequest
+            }.AsQueryable();
+
+            Expression<Func<LeaseRequest, bool>> predicate = t => t.Id == winnerLeaseRequest.Id;
```

Comment:
Why do we need that?

Reply:
fixed now. Found a simpler way to handle this test